traits/tests/test_historical_unpickling.py

```diff
@@ -0,0 +1,48 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+"""
+Tests for unpicking of pickles created using previous versions
+of Traits.
+"""
+
+import glob
+import os
+import pickle
+import unittest
+
+import pkg_resources
+
+from traits.api import BaseFloat
+
+
+def find_test_pickles():
+    """
+    Iterate over the pickle files in the test_data directory.
+
+    Yields paths to pickle files.
+    """
+    pickle_directory = pkg_resources.resource_filename(
+        "traits.tests",
+        "test-data/historical-pickles",
+    )
+
+    for filename in glob.glob(os.path.join(pickle_directory, "*.pkl")):
```

Comment:
`pathlib` makes this sort of wrangling nicer:
```
pickle_directory = Path(...)
for path in pickle_directory.glob("*.pkl"):
     yield path
```
and then you can use a `Path` object in the tests below.

Reply:
Done here, but not in `generate_pickles.py`: we might still want to run that script on Python 2.